packages/graphql-hooks/src/useClientRequest.js

```diff
@@ -60,13 +60,14 @@ function useClientRequest(query, initialOpts = {}) {
   const client = React.useContext(ClientContext)
   const isMounted = React.useRef(true)
   const activeCacheKey = React.useRef(null)
-  const operation = {
+  const operation = React.useRef(null)
+  operation.current = {
```

Comment:
Stupid question (still getting my head around useRef): Why not pass the object as the initial value, instead of `null`?

Reply:
By assigning `operation` on each render we make sure that inside `useCallback` we always have the latest version of `operation`.

`const operation = React.useRef({ ... })` will capture `query` and `initialOpts` only from the first render and may introduce bugs if these variables are changing between renders